pkg/controller/operators/olm/operator.go

```diff
@@ -553,11 +625,77 @@ func (a *Operator) transitionCSVState(in v1alpha1.ClusterServiceVersion) (out *v
 			return
 		}
 	} else {
-		// TODO: This should never be the case.
+		// This should never be the case
 		logger.Info("no targets annotation defined for CSV")
 		out.SetPhaseWithEvent(v1alpha1.CSVPhaseFailed, v1alpha1.CSVReasonNoTargetNamespaces, err.Error(), now, a.recorder)
 	}
 
+	// Check for intersecting provided APIs in intersecting OperatorGroups
+	options := metav1.ListOptions{
+		FieldSelector: fmt.Sprintf("metadata.name!=%s,metadata.namespace!=%s", operatorGroup.GetName(), operatorGroup.GetNamespace()),
+	}
+	otherGroups, err := a.client.OperatorsV1alpha2().OperatorGroups(metav1.NamespaceAll).List(options)
+
+	groupSurface := resolver.NewOperatorGroup(*operatorGroup)
+	otherGroupSurfaces := resolver.NewOperatorGroupSurfaces(otherGroups.Items...)
+
+	operatorSurface, err := resolver.NewOperatorFromCSV(out)
+	if err != nil {
+		// TODO: Add failure status to CSV
+		syncError = err
+		return
+	}
+	providedAPIs := operatorSurface.ProvidedAPIs().StripPlural()
+
+	switch result := a.apiReconciler.Reconcile(providedAPIs, groupSurface, otherGroupSurfaces...); {
```

Comment:
This switch encapsulates CSV failure transitions and provided API annotations for intersecting OperatorGroups.